### DISCUSSION TITLE
How to manage custom user roles for users? I only see "authenticated" user role is available. I want more roles.

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#3129] [Questions] [answered]
I don't see any provision for managing cutom user roles in superbase. Should i do it manually using profile table?

**Comment by steve-chavez** [comment]
You can add more roles, but for now our Kong layer only supports the hardcoded `authenticated`, `anon` and `service_role` roles.

Check this previous discussion: https://github.com/supabase/supabase/discussions/3107#discussioncomment-1298255

**Comment by awalias** [comment, marked as answer]
There is a good example of role-based access control here: https://github.com/supabase/supabase/blob/master/examples/nextjs-slack-clone/README.md#supabase-details

**Reply by Jaaneek** [reply]
> There is a good example of role-based access control here: https://github.com/supabase/supabase/blob/master/examples/nextjs-slack-clone/README.md#supabase-details

The link seems to not be working any more, can you update it? Thanks!

**Reply by GaryAustin1** [reply]
Not sure what that one was, but this is a pretty good repository on custom roles...
https://github.com/supabase-community/supabase-custom-claims

**Reply by Jaaneek** [reply]
I think I will go with new tables and reference user id's there, seems like much better way to manage permissions.

**Reply by albertonii** [reply]
I think this is the working repo @Jaaneek sent:
[Supabase-nextjs-slack-clone](https://github.com/antosan/supabase-nextjs-slack-clone/)

**Reply by rommyarb** [reply]
> There is a good example of role-based access control here: https://github.com/supabase/supabase/blob/master/examples/nextjs-slack-clone/README.md#supabase-details

@awalias 
The link returns 404 page not found. Is it moved somewhere? Thanks.

**Reply by zpuckeridge** [reply]
> > There is a good example of role-based access control here: https://github.com/supabase/supabase/blob/master/examples/nextjs-slack-clone/README.md#supabase-details
> 
> @awalias The link returns 404 page not found. Is it moved somewhere? Thanks.

https://github.com/supabase/supabase/tree/master/examples/slack-clone/nextjs-slack-clone#supabase-details

**Comment by cliveportman** [comment]
I have a table for each user role. When I create a user, I pass something in the meta_data called 'access_level'. Then I have a trigger function on a new user, which checks this access_level and adds the user to the relevant table. You can view the trigger function here:
https://github.com/supabase/supabase/discussions/8383

**Comment by ChuckJonas** [comment]
looks like supabase now supports adding custom roles via the UI, but there seems to be no official documentation around this...

**Reply by GaryAustin1** [reply]
The Roles UI is not really useful for adding roles to be used thru PostgrREST.  It lets you create a role and look at your roles.  There is SQL involved in getting a role to work thru the API.  This discussion covers some of that https://github.com/orgs/supabase/discussions/11948#discussioncomment-6783266

I've not seen anything from Supabase officially supporting additional roles or not especially around the role column in auth.users.   It does work though.
